src/frontends/lean/pp.cpp

```diff
@@ -447,6 +447,27 @@ bool pretty_fn::is_implicit(expr const & f) {
     }
 }
 
+bool pretty_fn::is_default_arg_app(expr const & e) {
+    if (m_implict || m_preterm)
+        return false; // showing default arguments
+    if (!closed(app_fn(e))) {
+        // the Lean type checker assumes expressions are closed.
+        return false;
```

Comment:
This is fine. We rarely try pretty print expressions containing de-Bruijn indices.